### PR TITLE
feat: add upcoming auctions rails to the home page

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [CX-3228] 

### Description

This PR adds upcoming auction results rails to the home page.

https://user-images.githubusercontent.com/11945712/208906298-eeae1cfb-a8fc-4968-9eef-9b3d0dde7246.mov

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes. // will add with the home screen ticket
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add upcoming auction results rail - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3228]: https://artsyproduct.atlassian.net/browse/CX-3228?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Home/Components/HomeUpcomingAuctionsRail.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  if (!meRes?.upcomingAuctionResults || meRes?.upcomingAuctionResults?.totalCount === 0) {
+    return null
+  }
+  return (
+    <Flex mb={mb}>
+      <Flex pl="2" pr="2">
+        <SectionTitle
+          title={title}
+          onPress={() => {
+            // do nothing
```
Comment (MounirDhahri) [inline src/app/Scenes/Home/Components/HomeUpcomingAuctionsRail.tsx:38]: will implement with the next ticket

File: src/app/Scenes/Home/Home.tsx [hunk 2]
```diff
@@ -200,6 +207,7 @@ const Home = (props: Props) => {
       title: "Similar to Works You've Viewed",
       type: "artwork",
       data: homePageBelow?.similarToRecentlyViewedArtworkModule,
+      hidden: !showUpcomingAuctionResultsRail,
```
Comment (MrSltun) [inline src/app/Scenes/Home/Home.tsx:210]: **Question:** is the new feature flag supposed to hide "Similar to Works You've Viewed"? 👀

**Review by MrSltun — DISMISSED** [review]
Looks good! 🌟 I just left one question

File: src/app/Scenes/Home/Home.tsx (hunk 2 above)
Comment (MounirDhahri) [reply src/app/Scenes/Home/Home.tsx:210]: aha 😵‍💫
thanks for the catch 😄

Comment (MounirDhahri) [reply src/app/Scenes/Home/Home.tsx:210]: I will fix

**Pull request merged**